state/factory/workingset.go

```diff
@@ -368,32 +340,52 @@ func (ws *workingSet) Process(ctx context.Context, actions []action.SealedEnvelo
 }
 
 func (ws *workingSet) process(ctx context.Context, actions []action.SealedEnvelope) error {
-	var err error
+	if err := ws.validate(ctx); err != nil {
+		return err
+	}
```

Comment:
I won't recommend modifying the order which may introduce another bug in a bug fix pr.

Reply:
yeah, should minimize the fix, see comment above

Reply:
> I won't recommend modifying the order which may introduce another bug in a bug fix pr. Or you may give more details in the description why this needs to be modified

IMO, the error for a false workingset should be returned as early as possible(as done in L408). Computation sources couldn't be wasted on validating an useless workingset. Maybe a fullsync should be done for this change?